JumpscaleCore/servers/gedis/handlers.py

```diff
@@ -300,21 +302,35 @@ def _handle_request(self, request, address, user_session):
         elif request.command.command == "auth":
             tid, seed, signature = request.arguments
             tid = int(tid)
-            try:
-                tclient = j.clients.threebot.client_get(threebot=tid)
-            except Exception as e:
-                logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
-                return (logdict, None)
-            try:
-                verification = tclient.verify_from_threebot(seed, signature)
-            except Exception as e:
-                logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
-                return (logdict, None)
-            # if we get here we know that the user has been authenticated properly
-            user_session.threebot_id = tclient.tid
-            user_session.threebot_name = tclient.name
-            j.shell()
 
+            current_threebot_id = int(j.tools.threebot.me.default.tid)
+            # If working on same machine no need to get a client to authenticate
+            # otherwise, we'll have infinite loop
+            if current_threebot_id != tid:
+                try:
+                    tclient = j.clients.threebot.client_get(threebot=tid)
+                except Exception as e:
+                    logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
+                    return (logdict, None)
+                try:
+                    verification = tclient.verify_from_threebot(seed, signature)
+                except Exception as e:
+                    logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
+                    return (logdict, None)
+
+                # if we get here we know that the user has been authenticated properly
+                user_session.threebot_id = tclient.tid
+                user_session.threebot_name = tclient.name
+            else:
+                # can't reuse verification methods in 3 bot client, otherwise we gonna go into infinite loop
+                # so we verify directly using nacl
+                try:
+                    VerifyKey(binascii.unhexlify(j.tools.threebot.me.default.nacl.verify_key_hex)).verify(seed, binascii.unhexlify(signature))
+                except Exception as e:
+                    logdict = j.core.myenv.exception_handle(e, die=False, stdout=True)
+                    return (logdict, None)
+                user_session.threebot_id = tid
+                user_session.threebot_name = j.tools.threebot.me.default.name
```

Comment:
This is not correct, this will always have the value `default`, since that's the name of the instance, this is not the threebot name, the threebot name is `j.tools.threebot.me.default.tname`.